### ISSUE TITLE
Version 15.0 Confconsole requires python-requests

### ISSUE THREAD

**Issue by Dude4Linux** [turnkeylinux/tracker#1066]
After building the latest version 15.0-rc1 of the TKLDev appliance and testing it in a Proxmox VM, the install fails to complete firstboot.  On a hunch, I checked confconsole and got the following errors.
```
root@tkldev-15 ~# confconsole 
Traceback (most recent call last):
  File "/usr/bin/confconsole", line 664, in <module>
    main()
  File "/usr/bin/confconsole", line 640, in main
    pm = plugin.PluginManager(PLUGIN_PATH, {'eventManager': em, 'interactive': interactive})
  File "/usr/lib/confconsole/plugin.py", line 168, in __init__
    current_plugin = Plugin(file_path, module_globals)
  File "/usr/lib/confconsole/plugin.py", line 71, in __init__
    self.module = imp.load_module(self.module_name, module_fob, module_pathname, module_description)
  File "/usr/lib/confconsole/plugins.d/Lets_Encrypt/get_certificate.py", line 3, in <module>
    import requests
ImportError: No module named requests
```
After installing python-requests and python-urllib3 confconsole functions normally
```
# apt-get install python-requests
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  python-urllib3
Suggested packages:
  python-cryptography python-idna python-openssl python-socks python-ntlm
Recommended packages:
  python-cryptography python-idna python-ipaddress python-openssl
The following NEW packages will be installed:
  python-requests python-urllib3
0 upgraded, 2 newly installed, 0 to remove and 4 not upgraded.
Need to get 178 kB of archives.
After this operation, 833 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
```
Although I've only experienced this with the TKLDev appliance, but I suspect the issue is shared by all v15 appliances.

**Comment by Dude4Linux** [comment]
Nevermind.  I just build and tested a 15.0-rc1 core appliance and did not experience the problem.  This issue must be due to an anomaly with the way I created the TKLDev appliance.

**Comment by Dude4Linux** [comment]
@JedMeister - I think you are probably correct.  I've been doing development while on the road using an LXD container I built following your guide at https://gist.github.com/JedMeister/ad6f3b405a889b62985dc933431f7dd1.
There I downloaded the latest TKLDev from github and for the first time was able to get the build to complete. (always failed before because of missing useraufs)  I then moved the product.iso to my Proxmox server and launched it as a new VM.  Since my process for creating LXD container images strips out all unnecessary packages including confconsole (not needed in containers).  I'll double check to make sure my container TKLDev is up-to-date.

**Comment by Dude4Linux** [comment]
@JedMeister - That was it.  I found that common was about two weeks out-of-date.  I see that you recently merged the stretch-dev branch into master, so from now on we should be using master instead of stretch-dev, right?  In any case, I switched buildtasks, common, and core to the master branch and updated each one to current.  Then rebuilt TKLDev and tested the product.iso in a Proxmox VM.  First boot ran as expected and confconsole works.  I think my previous attempt was hanging on the attempt to download the stretch bootstrap.
Now that we're using overlay(fs) instead of useraufs, I should be able to run this in an unprivileged container.

**Comment by JedMeister** [comment]
> I see that you recently merged the stretch-dev branch into master, so from now on we should be using master instead of stretch-dev, right?

Yes that's right. Apologies on that. I probably should have broadcast that more broadly!

> Now that we're using overlay(fs) instead of useraufs, I should be able to run this in an unprivileged container.

Hopefully! Let me know how that goes.

FWIW, The rc1 of Core and TKLDev are on the mirror. I plan to announce it via the blog today. :smile:

**Comment by Dude4Linux** [comment]
Unfortunately it appears TKLDev will continue to require a privileged container.  When I ran tkldev-setup in an unprivileged container it failed because mknod is not allowed.
```
# tar -zxf bootstrap-stretch-amd64.tar.gz -C stretch
tar: ./dev/ptmx: Cannot mknod: Operation not permitted
tar: ./dev/zero: Cannot mknod: Operation not permitted
tar: ./dev/full: Cannot mknod: Operation not permitted
tar: ./dev/tty: Cannot mknod: Operation not permitted
tar: ./dev/urandom: Cannot mknod: Operation not permitted
tar: ./dev/null: Cannot mknod: Operation not permitted
tar: ./dev/random: Cannot mknod: Operation not permitted
tar: Exiting with failure status due to previous errors
```
Stéphane Graber says 
> As for allowing mknod inside a user namespace, this simply isn’t going to happen as it would be a major security issue. If you were allowed to mknod things, you could mknod existing devices, say sda and then have an unprivileged user mess with that and wipe the disk on the host.

I don't know if it would be possible to just keep or add CAP_MKNOD and then use apparmor profile to limit it to the bootstrap directory.

**Comment by JedMeister** [comment]
> Unfortunately it appears TKLDev will continue to require a privileged container. When I ran tkldev-setup in an unprivileged container it failed because mknod is not allowed.

Hmm, I wonder if there is a way we could work around that within a container? Perhaps instead of mknod we could somehow mount bind? Using overlayFS perhaps?

Another thing that I probably need to flag with you too is that in v15.0 TKLDev, because OverlayFS lacks the full functionality provided by auFS (can't have infinite read/write layers), deck doesn't quite work the same as it used to. Actually, Alon rewrote it as a [bash script](https://github.com/turnkeylinux/deck/blob/master/deck).

**Comment by Dude4Linux** [comment]
@JedMeister - Yes, I did notice that `deck` was behaving differently in v15.0 TKLDev.  Specifically root.patched doesn't stay mounted after `make` has completed.  I had to add a test to lxd-make-image to check if it was mounted and if not, then remount it.  I had seen this behavior before when my laptop was shutdown overnight.  The `deck` mounts were not restored during reboot.  The change I made should take care of both cases, so be sure you grab the latest `lxd-make-image` from https://github.com/Dude4Linux/turnkey-pde/blob/master/lxd-make-image.

**Comment by Dude4Linux** [comment]
@JedMeister - I tried using `mount --bind /dev/ptmx stretch/dev/ptmx` for each of the missing device nodes, the `make` continued to fail.  First it was no overlay module found, which I fixed by adding overlay to the `linux.kernel_modules: overlay` in the container config. Then it was can't mount devpts in a chroot. I gave up for now trying to make it work in an unprivileged container.

I just ran
```
lxc init turnkey-tkldev-15.0-rc1-stretch-amd64-lxd tkldev-test -c security.privileged=true
lxc file push files/my_inithooks.conf tkldev-test/etc/inithooks.conf
lxc file push files/inithooks.service tkldev-test/lib/systemd/system/
lxc start tkldev-test
lxc exec tkldev-test /bin/bash
```
waited for initialization to complete and then ran core make successfully.
